src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -3326,29 +3326,48 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
                     x.base.base.loc);
             }
         }
-        LCOMPILERS_ASSERT(
-            ASRUtils::check_equal_type(ASRUtils::expr_type(lhs), ASRUtils::expr_type(rhs)));
```

Comment:
Just replace this with an `if` statement that gives error when types are not equal.